Algorithm/QCAlgorithm.Trading.cs

```diff
@@ -922,7 +909,18 @@ into ticket
             var quantity = CalculateOrderQuantity(symbol, percentage) - marketOrdersQuantity;
             if (Math.Abs(quantity) > 0)
             {
-                MarketOrder(symbol, quantity, false, tag);
+                var anyNonDailySubscriptions = security.Subscriptions.Any(x => x.Resolution != Resolution.Daily);
+
+                //Check whether the exchange is open and intraday data is available to send a market order. 
+                //If not, send a market on open order instead
+                if (security.Exchange.ExchangeOpen && anyNonDailySubscriptions)
```

Comment:
nice check, this makes sense to me. if exchange is open and we have intraday data, submit `MarketOrder` -- otherwise do `MarketOnOpen` -- love it.